docs/design/selector-generation.md

```diff
@@ -110,18 +110,42 @@ No changes
 ## Kubectl
 
 No required changes.
-Suggest moving SELECTOR to wide output of `kubectl get jobs` since users don't write the selector.
+Suggest moving SELECTOR to wide output of `kubectl get jobs` since users do not write the selector.
 
 ## Docs
 
 Remove examples that use selector and remove labels from pod templates.
 Recommend `kubectl get jobs -l job-name=name` as the way to find pods of a job.
 
-# Cross Version Compat
+# Conversion
 
-`v1beta1` will not have a `job.spec.manualSelector` and will not provide a default selector.
+The following applies to Job, as well as to other types that adopt this pattern.
+
+Type `extensions/v1beta1` will not have a `job.spec.manualSelector`.  Both the internal type and the `batch/v1` type will have
+`job.spec.manualSelector`.
+
+Since `extensions/v1beta1` will not have a `job.spec.manualSelector`, and since an object created via the `batch/v1` API endpoint
+might be stored internally as type `extensions/v1beta1`, there has to be a place to store that value.  It will be stored in an annotation.
+The annotation will be called `kubernetes.io/manual-selector`.
+
+- `extensions/__internal` to `extensions/v1beta1`: the value of `internal.ManualSelector` is stored in an annotation on the v1beta1 type.
+- `extensions/v1beta1` to `extensions/__internal`: the value of `internal.ManualSelector` is recovered from the annotation on the v1beta1 type.
+
+We would like the conversion to also have the following properties:
+
+1. Users that previously used v1beta1 do not start seeing a new annotation when they get back objects.
```

Comment:
As Brian says, only possible by omitting the annotation if it's empty. If set, there's no way to make it round-trippable AND invisible to users.